silverback/cluster/types.py

```diff
@@ -0,0 +1,281 @@
+    """Total managed secrets"""
+
+    @field_validator("cpu", mode="before")
+    def parse_cpu_value(cls, value: str | int) -> int:
+        if not isinstance(value, str):
+            return value
+
+        return round(math.log2(float(value.split(" ")[0]) * 1024 / 256))
+
+    @field_validator("memory", mode="before")
+    def parse_memory_value(cls, value: str | int) -> int:
+        if not isinstance(value, str):
+            return value
+
+        mem, units = value.split(" ")
+        if units.lower() == "mib":
+            assert mem == "512"
+            return 0
+
+        assert units.lower() == "gb"
+        return int(mem)
+
+    @field_validator("storage", mode="before")
+    def parse_storage_value(cls, value: str | int) -> int:
+        if not isinstance(value, str):
+            return value
+
+        storage, units = value.split(" ")
+        if units.lower() == "gb":
+            assert storage == "512"
+            return 0
+
+        assert units.lower() == "tb"
+        return int(storage)
+
+    @staticmethod
+    def _decode_byte(value: int, byte: int) -> int:
+        # NOTE: All configuration settings must be uint8 integer values when encoded
+        return (value >> (8 * byte)) & (2**8 - 1)  # NOTE: max uint8
+
+    @classmethod
+    def decode(cls, value: int) -> "ClusterConfiguration":
+        """Decode the configuration from 8 byte integer value"""
+        if isinstance(value, ClusterConfiguration):
+            return value  # TODO: Something weird with SQLModel
+
+        # NOTE: Do not change the order of these, these are not forwards compatible
+        return cls(
+            version=cls._decode_byte(value, 0),
+            cpu=cls._decode_byte(value, 1),
+            memory=cls._decode_byte(value, 2),
+            networks=cls._decode_byte(value, 3),
+            bots=cls._decode_byte(value, 4),
+            triggers=5 * cls._decode_byte(value, 5),
+            storage=cls._decode_byte(value, 6),
+            secrets=cls._decode_byte(value, 7),
+        )
+
+    @staticmethod
+    def _encode_byte(value: int, byte: int) -> int:
+        return value << (8 * byte)
+
+    def encode(self) -> int:
+        """Encode configuration as 8 byte integer value"""
+        # NOTE: Do not change the order of these, these are not forwards compatible
+        return (
+            self._encode_byte(self.version, 0)
+            + self._encode_byte(self.cpu, 1)
+            + self._encode_byte(self.memory, 2)
+            + self._encode_byte(self.networks, 3)
+            + self._encode_byte(self.bots, 4)
+            + self._encode_byte(self.triggers // 5, 5)
+            + self._encode_byte(self.storage, 6)
+            + self._encode_byte(self.secrets, 7)
+        )
+
+
+class ClusterTier(enum.IntEnum):
+    """Suggestions for different tier configurations"""
+
+    PERSONAL = ClusterConfiguration(
+        cpu="0.25 vCPU",
+        memory="512 MiB",
+        networks=3,
```

Comment:
:eyes:

Reply:
Yes, we really need an answer for this before the beta. This is only paid configuration, so not relevant for the alpha

Reply:
Seems like 1 | 3 might be a more common divider than 3 | 10. Maybe I've been in a hole but I've never had a product interact with more than 2 chains. Excluding staging/prod testnet/mainnet splits. Just a thought.

Reply:
In the past two years, having natively multichain products is pretty much the standard. I think it's a very common product expectation.

Reply:
I would expect to have a "testnet cluster" and a "prod cluster" as being the split, vs. mixing testnets and mainnets for a single ecosystem per cluster